src/types/index.ts

```diff
@@ -1,46 +1,408 @@
-// @ts-ignore
-interface AutoComplete {}
+// -----------------------------------------------------------------------------
+//  --------------------------------- TYPES ------------------------------------
+// -----------------------------------------------------------------------------
 
-interface StopGroup {}
+type GroundPlacesObject = Record<Gpuid, StopGroup | StopCluster>;
 
-interface StopCluster {}
+type Gpuid = string;
 
-interface StopGroupProperties {}
+type StopGroupGpuid = Gpuid;
 
-interface StopClusterProperties {}
+type StopClusterGpuid = Gpuid;
 
-interface Filters {}
+// -----------------------------------------------------------------------------
+//  ------------------------------- INTERFACES ---------------------------------
+// -----------------------------------------------------------------------------
 
-interface GroundPlacesDiff {}
+interface StopGroup {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.GROUP;
+  childs: SegmentProviderStop[];
+  serviced: Serviced;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type Gpuid = string;
+interface StopCluster {
+  country_code: CountryCode;
+  name: string;
+  longitude: number;
+  latitude: number;
+  type: GroundPlaceType.CLUSTER;
+  childs: StopGroupGpuid[];
+  serviced: Serviced;
+  unique_name?: string;
+  has_been_modified?: boolean;
+  warning?: boolean;
+  is_latest?: boolean;
+}
 
-type StopGroupGpuid = Gpuid;
+interface StopGroupInfos {
```

Comment:
Same here why a specific type for "infos" ?
Naming is wrong.

You should factorise your types.

Reply:
Same answer as before, to make distinction between them if their structures were be different

Reply:
After call, this is now grouped